pkgs/top-level/all-packages.nix

```diff
@@ -11298,6 +11298,14 @@ let
 
   cutecom = callPackage ../tools/misc/cutecom { };
 
+  cutegram =
+    let f = ../applications/networking/instant-messengers/telegram;
+        cp = qt5Libs.callPackage;
+    in cp "${f}/cutegram" rec {
```

Comment:
I’m not sure how this is any different from replacing every `cp` with `callPackage` and every `f` with  ../applications/networking/instant-messengers/telegram. Isn’t nix referentially transparent? Also, I’m not using `import` anywhere.
I don’t really understand what goes wrong here.

I chose this semi-top-level style because these libraries are only really used by cutegram in nixpkgs at the moment, but there might be other packages who use it, and since they are only libraries I don’t want to clutter the top-level.